scripts/wavie.js

```diff
@@ -0,0 +1,32 @@
+// Description
+//   Wave people as they enter and exit the room
+
+module.exports = function (robot) {
+    // Wave at the latest message in the room
+    function wave(room) {
+        // Make sure we have access to all the clients we need
+        if (!robot.adapter.client || !robot.adapter.client.web) {
+            return;
+        }
+
+        // Shorten web client so we can have nice concise lines :)
+        webClient = robot.adapter.client.web;
+
+        // Try channel as a public channel search first, then as a private channel
+        webClient.channels.info(room)
```

Comment:
We should be able to get the room from the original thing passed into the `robot.enter`/`robot.leave` callback.

Reply:
I don't understand-- that's exactly what i'm passing to wave() ?

Reply:
Apparently I'm bad with words atm - I meant we should probably be able to determine if it's public or private from there

Reply:
how so? I believe we're just given a room id-- is it determinable by the prefix letter or sum shit?

Reply:
Eh, if not that's why I said "probably" - should be able to do it nicer somehow but happy to leave it as is

Reply:
I investigated and room id indeed has a prefix letter that is used for distinguishing. 'G' is for groups (private), 'C' is for channels (public) and 'D' is for direct messages. I'll do a prefix check :)